### PR TITLE
:seedling: Unit tests for Policy.go

### PR DESCRIPTION
**Pull request by naveensrinivasan**
- Included tests for policy.go

#### What kind of change does this PR introduce?

(Is it a bug fix, feature, docs update, something else?)

- [ ] PR title follows the guidelines defined in our [pull request documentation](https://github.com/ossf/scorecard/blob/main/CONTRIBUTING.md#pr-process)

#### What is the new behavior (if this is a feature change)?**

- [ ] Tests for the changes have been added (for bug fixes/features)

#### Does this PR introduce a user-facing change?

For user-facing changes, please add a concise, human-readable release note to
the `release-note`

(In particular, describe what changes users might need to make in their
application as a result of this pull request.)

```release-note

```

### CODE REVIEW

File: .codecov.yml [hunk 1]
```diff
@@ -5,6 +5,7 @@ codecov:
     require_ci_to_pass: yes
 
 ignore:
+  - "policy/policy.pb.go"
```
Comment (spencerschrock) [inline .codecov.yml:8]: Should this be generalized to any `*.pb.go` file?

File: policy/policy_test.go [hunk 2]
```diff
@@ -131,3 +135,231 @@ func TestPolicyRead(t *testing.T) {
...
+	}
+
+	// Test panic with an unknown mode
+	testPanic := func() {
+		defer func() {
+			if r := recover(); r == nil {
+				t.Errorf("Expected panic, but no panic occurred")
+			}
+		}()
+		modeToProto("unknown")
```
Comment (spencerschrock) [inline policy/policy_test.go:297]: Unrelated to this change (feel free to resolve) but having modeToProto panic seems wrong.

File: policy/policy_test.go [hunk 3]
```diff
@@ -131,3 +135,231 @@ func TestPolicyRead(t *testing.T) {
...
+			requiredRequestTypes:  []checker.RequestType{checker.FileBased, checker.CommitBased},
+			expectedEnabledChecks: 0,
+			expectedError:         true,
+		},
+		{
+			name:                  "Without ScorecardPolicy and argsChecks",
+			sp:                    nil,
+			argsChecks:            []string{},
+			requiredRequestTypes:  []checker.RequestType{checker.FileBased, checker.CommitBased},
+			expectedEnabledChecks: 4,
```
Comment (spencerschrock) [inline policy/policy_test.go:338]: is it worth clarifying where this number comes from in a comment?

**Review by spencerschrock — APPROVED** [review]

File: policy/policy_test.go (hunk 3 above)
Comment (naveensrinivasan) [reply policy/policy_test.go:338]: I will do that.

File: policy/policy_test.go (hunk 2 above)
Comment (naveensrinivasan) [reply policy/policy_test.go:297]: I agree. Another PR.

File: .codecov.yml (hunk 1 above)
Comment (naveensrinivasan) [reply .codecov.yml:8]: I agree. Thanks

**Pull request merged**